test/exec.js

```diff
@@ -19,20 +19,16 @@ assert.ok(result.code > 0);
 
 // Test 'fatal' mode for exec, temporarily overriding process.exit
 var old_fatal = shell.config.fatal;
-var old_exit = process.exit;
-
-var exitcode = 9999;
-process.exit = function (_exitcode) {
-    exitcode = _exitcode;
-};
 
 shell.config.fatal = true;
 
-var result = shell.exec('asdfasdf'); // could not find command
-assert.equal(exitcode, 1);
+assert.throws(function() {
+  shell.exec('asdfasdf'); // could not find command
+}, function(err) {
+  return err.message === 'exec: internal error';
```

Comment:
Two things:
1. I'm pretty sure you can just pass in the expected error message here, and it will compare.
2. Is there any way to make the error message a little more descriptive?